thorium/resources.py

```diff
@@ -254,8 +259,10 @@ def _set(self, field_name, value, cast=False):
         field = self._fields[field_name]
         if not self._partial and not field.is_readonly and value == NotSet:
             raise errors.ValidationError('Attempted to set field {0} of a non-partial resource to NotSet'.format(field))
+        if not field.is_mutable and self.is_set(field_name) and self._get(field_name) != (field.default or value):
+            raise errors.ValidationError('Attempted to change value of non-mutable field.')
```

Comment:
The error message here should provide more information, like which field this was attempted on. See above error message for an example.